### PR TITLE
fix backend info issue

### PR DESCRIPTION
**Pull request by cqc-melf**
same as https://github.com/CQCL/pytket-qulacs/pull/4

### CODE REVIEW

File: pytket/extensions/stim/backends/stim_backend.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def __init__(self) -> None:
 
     @property
     def backend_info(self) -> Optional[BackendInfo]:
-        return self._backend_info
+        return None
```
Comment (cqc-alec) [inline pytket/extensions/stim/backends/stim_backend.py:142]: Should probably fully revert the change in the other PR?

**Review by cqc-alec — CHANGES_REQUESTED** [review]

File: pytket/extensions/stim/backends/stim_backend.py (hunk 1 above)
Comment (cqc-melf) [reply pytket/extensions/stim/backends/stim_backend.py]: done
